canonical-web-and-design/maas-ui
# feat(ui): add device filter accordion to device list

## Done

- Added device filter accordion to device list (and placeholder searchbox)
- Added selectors and utils for filtering devices

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/devices
- Add some filters via the accordion, and check that the search text is updated and the devices are filtered properly (note you can't filter via the search box yet)

## Fixes

Fixes canonical-web-and-design/app-tribe#530

## Screenshot

![Screenshot 2021-11-25 at 13-41-18 Devices bolla MAAS](https://user-images.githubusercontent.com/25733845/143375929-a609d994-acf1-4fe3-a367-5e5159e13f5d.png)
